common/util.go

```diff
@@ -151,17 +150,17 @@ func GetConnectionAddr(ctx context.Context) string {
 	return from
 }
 
-// RatToFixed converts a big.Rat into a fixed point number represented as int64
-// according to supplied scale factor and maximum decimalplaces
-func RatToFixed(rat *big.Rat, scalingFactor int64, maxDecimalPlaces int) (int64, error) {
-	if rat == nil {
+// PriceToFixed converts a big.Rat into a fixed point number represented as int64
+// using a scaleFactor of 1000 resulting in max decimal places of 3
+func PriceToFixed(price *big.Rat) (int64, error) {
+	scalingFactor := int64(1000)
+	if price == nil {
 		return 0, fmt.Errorf("reference to rat is nil")
 	}
-	scaled := new(big.Rat).Mul(rat, big.NewRat(scalingFactor, 1))
-	s := scaled.FloatString(maxDecimalPlaces)
-	intPart, err := strconv.ParseInt(strings.Split(s, ".")[0], 10, 64)
-	if err != nil {
-		return 0, err
+	scaled := new(big.Rat).Mul(price, big.NewRat(scalingFactor, 1))
+	fp, _ := new(big.Float).SetRat(scaled).Int64()
+	if fp == 0 {
+		return 0, fmt.Errorf("Too many decimal places")
```

Comment:
I don't think we should return an error here since 0 is a valid fixed point number. Furthermore, the error string indicates that the decimal representation of the price passed in has "too many decimal places", but there could be other scenarios besides when `fp == 0` where the decimal representation of the price passed in has "too many decimal places" i.e. 1.666666666

Reply:
Fixed in https://github.com/livepeer/go-livepeer/pull/939/commits/91536e4e33b71ae4cc5bcf1078ddda29ec399bff